cassandra-leaves-app/src/pages/index.js

```diff
@@ -1,53 +1,67 @@
-import Link from "next/link";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import Card from '../components/Card';
 
 export default function Home({ leavesData }) {
-    const [filteredData, setFilteredData] = useState(leavesData || []); // Default to an empty array
+    const [filteredData, setFilteredData] = useState(leavesData || []);
     const [searchQuery, setSearchQuery] = useState("");
     const [currentPage, setCurrentPage] = useState(1);
-    const itemsPerPage = 10;
+    const [isDarkMode, setIsDarkMode] = useState(false);
+    const itemsPerPage = 9;
 
     const [formState, setFormState] = useState({ id: "", title: "", domain_name: "" });
     const [isEditing, setIsEditing] = useState(false);
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [successMessage, setSuccessMessage] = useState("");
+    const [deleteRecord, setDeleteRecord] = useState(null);
+    const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
```

Comment:
Great job implementing the state management for this component! I see you've used `useState` effectively, but here are a few tips to make the code more scalable and maintainable:

### 1. **Group Related State**
Instead of having separate state variables for closely related pieces of data (e.g., `isModalOpen`, `successMessage`, `isDeleteModalOpen`), consider grouping them into a single object. For example:
```tsx
const [modalState, setModalState] = useState({
  isOpen: false,
  isDelete: false,
  successMessage: "",
});
```
This simplifies the number of state variables and makes the code cleaner.

---

### 2. **Separate UI State from Data State**
UI-related states like `isDarkMode` and `isModalOpen` could be separated from data-specific states (like `filteredData` and `formState`). This makes it easier to distinguish between UI concerns and the data logic.

---

### 3. **Leverage Derived State**
Instead of managing `filteredData` independently, you can derive it directly from `leavesData` and `searchQuery` using `useMemo`. For example:
```tsx
const filteredData = useMemo(() => {
  return leavesData?.filter(item => item.title.includes(searchQuery)) || [];
}, [leavesData, searchQuery]);
```
This avoids the need to keep the `filteredData` state synchronized manually.

---

### 4. **Encapsulate Reusable Logic in Custom Hooks**
If you find yourself managing similar state patterns across multiple components (e.g., modal handling), consider creating a custom hook. For example:
```tsx
function useModal() {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen((prev) => !prev);

  return { isOpen, toggle, setIsOpen };
}
```
This makes your code more reusable and easier to maintain.

---

### 5. **Consider Context or Global State for Shared States**
States like `isDarkMode` might be used across multiple components. If so, moving them to a context or global state (like Redux or Zustand) could improve scalability:
```tsx
const DarkModeContext = createContext();

function DarkModeProvider({ children }) {
  const [isDarkMode, setIsDarkMode] = useState(false);

  return (
    <DarkModeContext.Provider value={{ isDarkMode, setIsDarkMode }}>
      {children}
    </DarkModeContext.Provider>
  );
}
```

---

### 6. **Document State Purpose**
Adding comments to explain the purpose of each state variable makes the code easier to understand and maintain. For example:
```tsx
// Tracks the current page for pagination
const [currentPage, setCurrentPage] = useState(1);
```

---

### 7. **Refactor When Complexity Grows**
As the component evolves, consider revisiting and refactoring the state logic. Breaking down the component into smaller, more focused components might also help reduce complexity.

Let me know if you’d like to pair on any of these or need examples to get started! These changes will not only improve this component but also help you write more scalable React code in the future.

Great work overall! Looking forward to the next iteration of this code. 🚀

---